### PR TITLE
[aot] [llvm] LLVM AOT Field part-4: Added AOT tests for Fields - CUDA backend

### PR DESCRIPTION
**Pull request by jim19930609**
Related issue = #4800

### CODE REVIEW

**Review by k-ye — APPROVED** [review]
Awesome!

nit on the title: Let's avoid using `#\d+` in the title, cause it could link to irrelevant issue/PRs... An alternative would be `\d+/n`

File: tests/cpp/backends/llvm/field_aot_test.cpp [hunk 1]
```diff
@@ -57,63 +37,115 @@ TEST(LlvmAOTTest, Field) {
...
+  std::stringstream aot_mod_ss;
+  aot_mod_ss << folder_dir;
+  aot_params.module_path = aot_mod_ss.str();
+  aot_params.program = &prog;
+  std::unique_ptr<aot::Module> mod = cpu::make_aot_module(aot_params);
+
+  run_field_tests(mod.get(), &prog, result_buffer);
+}
+
+TEST(LlvmAOTTest, CUDAField) {
```
[the remark this answers is not in the document]
Comment (jim19930609) [reply tests/cpp/backends/llvm/field_aot_test.cpp]: done, thx!

File: tests/cpp/backends/llvm/field_aot_test.cpp [hunk 2]
```diff
@@ -57,63 +37,115 @@ TEST(LlvmAOTTest, Field) {
...
+    ctx.runtime = prog->get_llvm_runtime();
     k_check_activate_pointer_fields->launch(&ctx);
   }
 
   // Check assertion error from ti.kernel
-  prog.check_runtime_error(result_buffer);
+  prog->check_runtime_error(result_buffer);
+}
+
+TEST(LlvmAOTTest, CPUField) {
```
[the remark this answers is not in the document]
Comment (jim19930609) [reply tests/cpp/backends/llvm/field_aot_test.cpp]: done, thx!

File: tests/cpp/backends/llvm/field_aot_cuda.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+from utils import compile_field_aot
+
+import taichi as ti
+
+compile_field_aot(arch=ti.cuda)
```
[the remark this answers is not in the document]
Comment (jim19930609) [reply tests/cpp/backends/llvm/field_aot_cuda.py]: done, thx!

**Comment by jim19930609** [comment]
> Awesome!
> 
> nit on the title: Let's avoid using `#\d+` in the title, cause it could link to irrelevant issue/PRs... An alternative would be `\d+/n`

Thanks, didn't realize this myself :) Will stick to the alternative solution and avoid using "#" in the tile.

**Pull request merged**